src/main/java/uk/gov/pay/directdebit/payments/services/TransactionService.java

```diff
@@ -123,9 +123,12 @@ public PaymentRequestEvent paymentCreatedFor(Transaction transaction, Payer paye
         return paymentRequestEventService.registerPaymentCreatedEventFor(updatedTransaction);
     }
 
-    public PaymentRequestEvent paymentFailedFor(Transaction transaction) {
-        Transaction newTransaction = updateStateFor(transaction, SupportedEvent.PAYMENT_FAILED);
-        return paymentRequestEventService.registerPaymentFailedEventFor(newTransaction);
+    public PaymentRequestEvent paymentFailedFor(Transaction transaction, Payer payer, boolean sendEmail) {
+        if (sendEmail) {
```

Comment:
I believe this kind of conditionals should be avoided as you need to carry around that boolean for every call of this method. I think a cleaner way is to split lines `130-131` into a separate method `paymentFailedWithoutEmailFor(...)` and to have line `128` and a call to `paymentFailedWithoutEmailFor(...)` into another method `paymentFailedWithEmailFor(...)`. Then you will call each method based on your situation.

Reply:
Good point!